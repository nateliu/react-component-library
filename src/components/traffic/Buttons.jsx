import React, { Component } from 'react';
import { GO, CAUTION, STOP } from '../../constants/traffic/actionTypes';
import './Buttons.less';

const Buttons = ({
    go,
    caution,
    stop,
    lightStatus
}) => {
    return (
        <div style={{ textAlign: 'center' }}>
            <button className='trafficButton' onClick={go}
                disabled={lightStatus == GO || lightStatus == CAUTION}>
                Go
            </button>

            <button className='trafficButton' onClick={caution}
                disabled={lightStatus == CAUTION || lightStatus == STOP}>
                Caution
            </button>

            <button className='trafficButton' onClick={stop}
                disabled={lightStatus == STOP || lightStatus == GO}>
                Stop
            </button>
        </div>
    )
}

export default Buttons;