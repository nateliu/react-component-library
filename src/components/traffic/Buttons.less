  .trafficButton {
    height: 30px;
    width: 80px;
    border-radius: 4px;
    margin-left: 10px;
    cursor: 'pointer'
  }