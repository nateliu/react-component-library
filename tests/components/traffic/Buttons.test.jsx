import React from 'react';
import { shallow } from 'enzyme';
import Buttons from '../../../src/components/traffic/Buttons';

function setup() {
    const props = {
        go :jest.fn(),
        caution:jest.fn(),
        stop: jest.fn(),
        lightStatus:"GO"
    }

    const enzymeWrapper = shallow(<Buttons {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('comment component', () => {
    it('should render correctly with default state', () => {
        const { enzymeWrapper } = setup()

        expect(enzymeWrapper).toMatchSnapshot();
    })
})