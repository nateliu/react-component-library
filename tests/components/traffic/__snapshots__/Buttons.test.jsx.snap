// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comment component should render correctly with default state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Buttons
    caution={[MockFunction]}
    go={[MockFunction]}
    lightStatus="GO"
    stop={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="trafficButton"
          disabled={true}
          onClick={[MockFunction]}
        >
          Go
        </button>,
        <button
          className="trafficButton"
          disabled={false}
          onClick={[MockFunction]}
        >
          Caution
        </button>,
        <button
          className="trafficButton"
          disabled={true}
          onClick={[MockFunction]}
        >
          Stop
        </button>,
      ],
      "style": Object {
        "textAlign": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Go",
          "className": "trafficButton",
          "disabled": true,
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Go",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Caution",
          "className": "trafficButton",
          "disabled": false,
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Caution",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Stop",
          "className": "trafficButton",
          "disabled": true,
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Stop",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="trafficButton"
            disabled={true}
            onClick={[MockFunction]}
          >
            Go
          </button>,
          <button
            className="trafficButton"
            disabled={false}
            onClick={[MockFunction]}
          >
            Caution
          </button>,
          <button
            className="trafficButton"
            disabled={true}
            onClick={[MockFunction]}
          >
            Stop
          </button>,
        ],
        "style": Object {
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Go",
            "className": "trafficButton",
            "disabled": true,
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Go",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Caution",
            "className": "trafficButton",
            "disabled": false,
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Caution",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Stop",
            "className": "trafficButton",
            "disabled": true,
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Stop",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
